tests/unit/db/account_tests.py

```diff
@@ -0,0 +1,600 @@
+                self.client.basic_auth_str(),
+                expected
+                )
+        finally:
+            self.client.disconnect()
+
+    def test_all_dbs(self):
+        """
+        Test getting a list of all of the databases in the account
+        """
+        try:
+            self.client.connect()
+            self.assertIsInstance(self.client.all_dbs(), list)
+        finally:
+            self.client.disconnect()
+
+    def test_create_delete_database(self):
+        """
+        Test database creation and deletion
+        """
+        try:
+            self.client.connect()
+            # Create database
+            db = self.client.create_database('unittestdb')
+            self.assertTrue(db.exists())
+            # Delete database
+            self.assertIsNone(self.client.delete_database('unittestdb'))
+            self.assertFalse(db.exists())
+        finally:
+            self.client.disconnect()
+
+    def test_create_existing_database(self):
+        """
+        Test creation of already existing database
+        """
+        try:
+            self.client.connect()
+            self.client.create_database('_replicator', throw_on_exists=True)
+            self.fail('Above statement should raise a CloudantException')
+        except CloudantException, err:
+            self.assertEqual(str(err), 'Database _replicator already exists')
+        finally:
+            self.client.disconnect()
+
+    def test_delete_non_existing_database(self):
+        """
+        Test deletion of non-existing database
+        """
+        try:
+            self.client.connect()
+            self.client.delete_database('no_such_db')
+            self.fail('Above statement should raise a CloudantException')
+        except CloudantException, err:
+            self.assertEqual(str(err), 'Database no_such_db does not exist')
+        finally:
+            self.client.disconnect()
+
+    def test_keys(self):
+        """
+        Test retrieving the list of database names for the given client account
+        """
+        try:
+            self.client.connect()
+            self.assertEqual(self.client.keys(), [])
+            self.assertEqual(
+                self.client.keys(remote=True),
+                self.client.all_dbs()
+                )
+        finally:
+            self.client.disconnect()
+
+    def test_get_non_existing_db_via_getitem(self):
+        """
+        Test __getitem__ when retrieving a non-existing database
+        """
+        try:
+            self.client.connect()
+            db = self.client['no_such_db']
+            self.fail('Above statement should raise a KeyError')
+        except KeyError:
+            pass
+        finally:
+            self.client.disconnect()
+
+    def test_get_db_via_getitem(self):
+        """
+        Test __getitem__ when retrieving a database
+        """
+        try:
+            self.client.connect()
+            self.client.create_database('unittestdb')
```

Comment:
It would be a good idea to uniquify these database names using a helper method - that way multiple parallel test runs can be done on the same instance without any conflicts occurring.